### PR TITLE
Some fixes to rich-presence-related error reporting

### PR DESCRIPTION
**Pull request by gdeOo**
1) Rich presence format parameters must be function calls. If anything else was
provided instead, a null object exception would bubble up to the user. An editor
error, "Invalid parameter, expected function call", is now reported to the user
instead.

2) In rich presence format strings, an empty placeholder resulted in a .net
exception bubbling up to the user. The tokenizer read an empty token inside the
brackets which was then passed onto Int32.Parse, causing the exception. There
are now 2 editor errors instead: if the brackets do not contain a number,
"Empty parameter index" is reported. Otherwise, if Int32.Parse fails,
"Invalid parameter index" is reported (can happen if the number is a float).

3) A minor bug that caused errors generated while processing a rich
presence lookup to not bubble-up to the editor. The lookup was silently not
generated.

### CODE REVIEW

File: Parser/Functions/RichPresenceDisplayFunction.cs [hunk 1]
```diff
@@ -89,15 +95,19 @@ private ParseErrorExpression ProcessRichPresenceDisplay(StringConstantExpression
                 }
                 tokenizer.Advance();
 
-                var parameterIndex = Int32.Parse(index.ToString());
-                if (parameterIndex >= varargs.Entries.Count)
+                Int32 parameterIndex;
+                if (!Int32.TryParse(index.ToString(), out parameterIndex)
+                    || parameterIndex >= varargs.Entries.Count)
```
Comment (Jamiras) [inline Parser/Functions/RichPresenceDisplayFunction.cs]: should probably also check for negative values

File: Parser/Functions/RichPresenceLookupFunction.cs [hunk 2]
```diff
@@ -52,7 +52,9 @@ public override ParseErrorExpression BuildMacro(RichPresenceDisplayFunction.Rich
             if (value == null)
                 return (ParseErrorExpression)result;
 
-            context.RichPresence.AddLookupField(name.Value, dictionary);
+            var err = context.RichPresence.AddLookupField(name.Value, dictionary);
+            if (err != null)
+                return err;
```
[the remark this answers is not in the document]
Comment (gdeOo) [reply Parser/Functions/RichPresenceLookupFunction.cs]: Got it, changed.

File: Parser/Functions/RichPresenceDisplayFunction.cs (hunk 1 above)
Comment (gdeOo) [reply Parser/Functions/RichPresenceDisplayFunction.cs]: Agreed, added negative value check.

**Comment by gdeOo** [comment]
Looks like github doesn't handle force pushes very well.. can't see the diff between the pre and post force push versions. I usually use gitlab, where it works fine, and I use it often to avoid creating new commits that just fix small things related to code review. The changes were trivial, but still, sorry about that.

**Pull request merged**